src/Elastic.Documentation.Configuration/Versions/VersionInference.cs

```diff
@@ -29,11 +43,59 @@ public VersioningSystem InferVersion(string repositoryName, IReadOnlyCollection<
 
 		return versioning;
 	}
+
+	private VersioningSystem? VersioningFromApplicability(ApplicableTo applicableTo)
+	{
+		VersioningSystem? versioning = null;
+		if (applicableTo.ProductApplicability is not null)
+		{
+			versioning = applicableTo.ProductApplicability switch
+			{
+				{ ApmAgentAndroid: not null } => VersionsConfiguration.VersioningSystems[VersioningSystemId.ApmAgentAndroid],
```

Comment:
Maybe we should have an exhaustive `ProductFromApplicability` and have a unit test to ensure all known products are handled.